### PR TITLE
Query-tee: don't try to compare non-JSON responses

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR fixes an issue where query-tee logs a cryptic error message when one of the response payloads being compared is not JSON.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [x] Tests updated
- [n/a] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by colega — APPROVED** [review]
LGTM, but I guess we'll need to do this at some point?

**Review by aknuds1 — APPROVED** [review]
LGTM

File: tools/querytee/proxy_endpoint.go [hunk 1]
```diff
@@ -218,17 +222,26 @@ func (p *ProxyEndpoint) compareResponses(expectedResponse, actualResponse *backe
 	}
 
 	if expectedResponse.status != actualResponse.status {
-		return fmt.Errorf("expected status code %d but got %d", expectedResponse.status, actualResponse.status)
+		return fmt.Errorf("expected status code %d (returned by preferred backend) but got %d from secondary backend", expectedResponse.status, actualResponse.status)
+	}
+
+	if expectedResponse.contentType != "application/json" {
+		return fmt.Errorf("skipped comparison of response because the response from the preferred backend contained an unexpected content type '%s', expected 'application/json'", expectedResponse.contentType)
```
Comment (pracucci) [inline tools/querytee/proxy_endpoint.go:229]: Returning an error here will get logged as "error" level and tracked to `cortex_querytee_responses_compared_total{result="fail"}`. I'm not sure that's correct. We can't compare, so we just don't know if the comparison would have succeeded or failed. I think we shouldn't log it as error and track it in the metric with a new reason, like `skipped`.

Comment (charleskorn) [reply tools/querytee/proxy_endpoint.go]: Makes sense to me, I've changed this in 5d63f01.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**